files/default/neutron-ha-tool.py

```diff
@@ -516,39 +523,39 @@ def replicate_dhcp(qclient, noop=False):
     return errors
 
 
-def migrate_l3_routers_from_agent(qclient, agent_id, target_ids, noop):
-    LOG.info("Querying agent_id=%s for routers to migrate away", agent_id)
-    router_id_list = list_routers_on_l3_agent(qclient, agent_id)
+def migrate_l3_routers_from_agent(qclient, agent, targets, noop, wait_for_ports):
+    LOG.info("Querying agent_id=%s for routers to migrate away", agent['id'])
+    router_id_list = list_routers_on_l3_agent(qclient, agent['id'])
 
     migrations = 0
     errors = 0
     for router_id in router_id_list:
-        target_id = random.choice(target_ids)
-        if migrate_router_safely(qclient, noop,
-                                 router_id, agent_id, target_id):
+        target = random.choice(targets)
+        if migrate_router_safely(qclient, noop, router_id,
+                                 agent, target, wait_for_ports):
             migrations += 1
         else:
             errors += 1
 
     return (migrations, errors)
 
 
-def migrate_router_safely(qclient, noop, router_id, agent_id, target_id):
+def migrate_router_safely(qclient, noop, router_id, agent, target, wait_for_ports):
     if noop:
         LOG.info("Would try to migrate router=%s from agent=%s "
-                 "to agent=%s", router_id, agent_id, target_id)
+                 "to agent=%s", router_id, agent['id'], target['id'])
         return True
 
     try:
-        migrate_router(qclient, router_id, agent_id, target_id)
+        migrate_router(qclient, router_id, agent, target, wait_for_ports)
         return True
     except:
```

Comment:
Not part of this PR, but this is too broad and we are the ones raising (some of) the exceptions so we should try to catch the 2 know raises that we emit and then fall back to catching everything.